pkg/shell-operator/operator.go

```diff
@@ -538,8 +546,25 @@ func (op *ShellOperator) TaskHandleHookRun(t task.Task) queue.TaskResult {
 		err = op.HookMetricStorage.SendBatch(result.Metrics, map[string]string{
 			"hook": hookMeta.HookName,
 		})
+		if err != nil {
+			goto Metrics
```

Comment:
Oh, we will definitely have to refactor this method in the future...

Reply:
Aye, the control flow is horrible here. In my defense, it was horrible before I arrived. :)